openedx/core/djangoapps/video_pipeline/api.py

```diff
@@ -29,10 +31,16 @@ def update_3rd_party_transcription_service_credentials(**credentials_payload):
     if pipeline_integration.enabled:
         try:
             video_pipeline_user = pipeline_integration.get_service_user()
+            oauth_client = Client.objects.get(name=pipeline_integration.client_name)
         except ObjectDoesNotExist:
             return error_response, is_updated
 
-        client = create_video_pipeline_api_client(user=video_pipeline_user, api_url=pipeline_integration.api_url)
+        client = create_video_pipeline_api_client(
```

Comment:
@nasthagiri You may want to look at this. 

**Context**: We had a meeting earlier regarding JWT secrets and it was mentioned that we won't be able to get the secrets. 

I did some more research on it, and I found that in [programs tasks](https://github.com/edx/edx-platform/blob/6d2468d0dcf782a7561ecc61946146d6fae2762f/openedx/core/djangoapps/programs/tasks/v1/tasks.py#L24) we are doing something interesting. We are **using JWT auth but using Oauth client secrets to build JWT token**. This is something that is useful to us too, since we have Oauth client for Veda as well so we can use those secrets to build JWT token too. This will have 2 benefits; 1) We don't have to change video pipleline, 2) Once Ansible role work is complete migration to it would be easy and straightforward.